prune/service.go

```diff
@@ -0,0 +1,208 @@
+package prune
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/newscred/webhook-broker/config"
+	"github.com/newscred/webhook-broker/storage"
+	"github.com/newscred/webhook-broker/storage/data"
+	"github.com/rs/zerolog/log"
+	"gocloud.dev/blob"
+)
+
+type ArchiveDirector struct {
+	LocalArchiveManager  *ArchiveWriteManager
+	RemoteArchiveManager *ArchiveWriteManager
+}
+
+func (director *ArchiveDirector) Close() {
+	if director.RemoteArchiveManager != nil {
+		err := director.RemoteArchiveManager.Close()
+		if err != nil {
+			log.Error().Err(err).Msg("failed to close remote archive manager")
+		}
+	}
+	err := director.LocalArchiveManager.Close()
+	if err != nil {
+		log.Error().Err(err).Msg("failed to close local archive manager")
+	}
+}
+
+func buildRemoteObjectName(pruneConfig config.MessagePruningConfig) string {
+	now := time.Now().UTC().Format("2006_01_02T15_04_05Z")
+	objectName := fmt.Sprintf("%s_%s.jsonl", pruneConfig.GetExportNodeName(), now)
+	if len(pruneConfig.GetRemoteFilePrefix()) > 0 {
+		objectName = fmt.Sprintf("%s/%s", pruneConfig.GetRemoteFilePrefix(), objectName)
+	}
+	return objectName
+}
+
+var (
+	initLocalArchiveManager = func(pruneConfig config.MessagePruningConfig) (*ArchiveWriteManager, error) {
```

Comment:
Just curious, why are these functions defined like this (why not `func initLocalArchiveManager`)?

Reply:
Defining function as a variable makes it easier to mock this function when we want to test this function returning an error.